File: code/python/isValidBST.py
    def isValidBST(self, root: TreeNode) -> bool:
        if root == None or (root.left==None and root.right==None):
            return True
        lst = []
        self.go(root,lst)
        for i in range(len(lst)-1):
            if lst[i] >= lst[i+1]:
                return False
        return True
    
    def go(self,root:TreeNode,lst:list) -> None:
        if root.left == None and root.right == None:
            lst.append(root.val)
            return
        if root.left != None:
            self.go(root.left,lst)
        lst.append(root.val)
        if root.right != None:
            self.go(root.right,lst)
        return

File: code/java/isValidBST.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public boolean isValidBST(TreeNode root) {
        if((root == null) || ((root.left == null) && (root.right == null)))
            return true;
        List<Integer> lst = new ArrayList<Integer>();
        helper(root,lst);
        for(int i=0;i<lst.size()-1;i++){
            if(lst.get(i) >= lst.get(i+1))
                return false;
        }
        return true;
    }
    
    public void helper(TreeNode root,List lst){
        if(root.left == null && root.right == null){
            lst.add(root.val);
            return ;
        }
        if(root.left != null)
            helper(root.left,lst);
        lst.add(root.val);
        if(root.right != null)
            helper(root.right,lst);
        return ;
    }
}



